adlfs/lib.py

```diff
@@ -32,8 +33,40 @@
 class DatalakeRESTException(IOError):
     pass
 
+default_tenant = os.environ.get('azure_tenant_id', "common")
+default_username = os.environ.get('azure_username', None)
+default_password = os.environ.get('azure_password', None)
+default_client = os.environ.get('azure_client_id', None)
+default_secret = os.environ.get('azure_client_secret', None)
+default_resource = "https://management.core.windows.net/"
+default_store = os.environ.get('azure_store_name', None)
+default_suffix = os.environ.get('azure_url_suffix', '')
 
-def auth(tenant_id, username, password):
+
+def refresh_token(token):
+    """ Refresh an expired authorization token
+
+    Parameters
+    ----------
+    token : dict
+        Produced by `auth()` or `refresh_token`.
+    """
+    if token.get('refresh', False) is False:
+        raise ValueError("Token cannot be aut-refreshed.")
+    context = adal.AuthenticationContext('https://login.microsoftonline.com/' +
+                                         token['tenant'])
+    out = context.acquire_token_with_refresh_token(token['refresh'],
```

Comment:
I think this should raise a reasonable exception if it fails without a try/except. I can check.